Android build fix

linusmartensson opened the pull request:
Since OS is transformed from android to linux, we need to ensure that conditions
specifying OS==linux are prevented from executing by additionally specifying
'and android_build=0'. 

This was done on most conditions in bnoordhuis patch, but I added a couple missed ones that broke the android build.

Nodejs-Jenkins commented:
Thank you for contributing this pull request! Here are a few pointers to make sure your submission will be considered for inclusion.

Commit linusmartensson/libuv@66f5097e7246452de4fdd35a789dcd464f4b923e has the following error(s):
- Commit message must indicate the subsystem this commit changes

You can fix all these things _without_ opening another issue.

Please see [CONTRIBUTING.md](https://github.com/joyent/node/blob/master/CONTRIBUTING.md) for more information

bnoordhuis commented on common.gypi:
```diff
@@ -143,7 +143,7 @@
           }],
-          [ 'OS=="linux"', {
+          [ 'OS=="linux" and android_build == 0', {
             'cflags': [ '-ansi' ],
```
What does -ansi break (and why)?

bnoordhuis commented on uv.gyp:
```diff
@@ -51,7 +51,7 @@
           }],
-          ['OS == "linux"', {
+          ['OS == "linux" and android_build == 0', {
             'defines': [ '_POSIX_C_SOURCE=200112' ],
```
Ditto. Should be harmless, right?

Commit:
Android build target resolve

Work around the build name issues by instead
manually adding the build target for android
builds using '-DOS=android'

This additionally resolves problems caused by the
OS variable being rewritten from "android" to
"linux" in gyp.
This causes errors, as real-time support (-lrt)
and pthread support (-lpthread) is not available
in the android NDK toolchain as libraries.
The functions present in these libraries are
included automatically during the build instead
during the android compilation process.

bnoordhuis commented:
I'm okay with the patch but please update the commit log. Wasn't there a Makefile fix-up as well? -lrt?

linusmartensson commented:
This is, I hope, a bit clearer.

bnoordhuis commented:
Thanks, landed in ff3ee84.
